### DISCUSSION TITLE
Welcome to ALPS Discussions!

### DISCUSSION THREAD

**Discussion by danielver02** [danielver02/ALPS#95] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MSAfify** [comment]
Hi Prof. Daniel,
After I executed this test "./run_test_suite.sh", I found some errors:
ALPS
Thu Jul 20 13:37:54 CEST 2023
ERRORS in ALPS Execution of test_kpar_fast
Thu Jul 20 13:37:56 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_kperp
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:37:56 CEST 2023
No Errors in generate_distribution execution of test_kperp
Thu Jul 20 13:37:56 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:37:56 CEST 2023
ERRORS in ALPS Execution of test_kperp
Thu Jul 20 13:37:59 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_ICW
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:37:59 CEST 2023
No Errors in generate_distribution execution of test_ICW
Thu Jul 20 13:37:59 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:37:59 CEST 2023
ERRORS in ALPS Execution of test_ICW
Thu Jul 20 13:38:01 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_map
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:38:01 CEST 2023
No Errors in generate_distribution execution of test_map
Thu Jul 20 13:38:01 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:38:01 CEST 2023
ERRORS in ALPS Execution of test_map
Thu Jul 20 13:38:03 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_relativistic
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:38:03 CEST 2023
No Errors in generate_distribution execution of test_relativistic
Thu Jul 20 13:38:04 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:38:04 CEST 2023
ERRORS in ALPS Execution of test_relativistic
Thu Jul 20 13:38:06 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_bimax
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:38:06 CEST 2023
No Errors in generate_distribution execution of test_bimax
Thu Jul 20 13:38:06 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:38:06 CEST 2023
ERRORS in ALPS Execution of test_bimax
Thu Jul 20 13:38:08 CEST 2023
-=-=-=-=-=-=-=

-=-=-=-=-=-=-=
Testing:  test_analytical
-=-=-=-=-=-=-=
Generate Distribution
Thu Jul 20 13:38:08 CEST 2023
No Errors in generate_distribution execution of test_analytical
Thu Jul 20 13:38:08 CEST 2023
-=-=-=-=-=-=-=
ALPS
Thu Jul 20 13:38:08 CEST 2023
ERRORS in ALPS Execution of test_analytical
Thu Jul 20 13:38:11 CEST 2023
ERRORS in ALPS Execution of Test Suite. Investigate...

Could you please figure out this issue?

Many thanks in advance!
Mahmoud

**Reply by danielver02** [reply]
Hi Mahmoud,

It seems there is a problem with the actual MPI execution of the code, not with the generation of the f0-tables. It may help if you post one of the .out files in the ./tests folder. This should give us an idea about where the ALPS code stops.

Best wishes,
-Daniel

**Reply by MSAfify** [reply]
Hi,
Here is the required file [https://drive.google.com/file/d/11hnNodGglQ2qyQm9F8jFWQYvUGgW7Go-/view?usp=sharing](url)

Great thanks!
Mahmoud

**Reply by MSAfify** [reply]
Hi, 
try that one https://drive.google.com/file/d/11hnNodGglQ2qyQm9F8jFWQYvUGgW7Go-/view?usp=sharing

**Reply by danielver02** [reply]
Hi Mahmoud,

That file is empty. This suggests that the code didn't even start running. Maybe you can enter the command line command from the test routine directly into your shell. Then you can see what error message is produced. It seems to be a problem even before ALPS starts. Maybe something with your MPI setup?

You could try the following (from within the ALPS main directory):
`mpirun -np 4 --oversubscribe ./src/ALPS tests/test_kpar_fast.in`
and see what output you get.

Best wishes,
-Daniel

**Reply by MSAfify** [reply]
Hi Prof. Daniel,
Here is the output
Starting ALPS===================================
Time:
2023-07-20 -- 15:24:25
================================================
 ===========================================================
 I                                                         I
 I                       A  L  P  S                        I
 I              Arbitrary Linear Plasma Solver             I
 I                                                         I
 I                       Version 1.0                       I
 I                                                         I
 I  Kristopher Klein   (kgklein@arizone.edu)               I
 I  Daniel Verscharen  (d.verscharen@ucl.ac.uk)            I
 I                                                         I
 ===========================================================
All processes are up and running.
Reading from Input File: test_kpar_fast
GUESS ROUTINE: 
Intial Guess   1 :     9.9000E-04   -5.5000E-10
SPECIES PARAMETERS: 
Species   1 : 
 ns/nREF =  1.0000E+00 | qs/qREF =  1.0000E+00 | ms/mREF =  1.0000E+00
 Number of fitted functions =    1
 Relativistic effects = F
Species   2 : 
 ns/nREF =  1.0000E+00 | qs/qREF = -1.0000E+00 | ms/mREF =  5.4466E-04
 Number of fitted functions =    1
 Relativistic effects = F
-=-=-=-=-=-=-=-=-
Species  1, function  1, Maxwellian fit: 
 Initial fit parameter  1 =     1.7959E-01
 Initial fit parameter  2 =     1.0000E+00
 Initial fit parameter  3 =     0.0000E+00
 Initial fit parameter  4 =     0.0000E+00
 Initial fit parameter  5 =     0.0000E+00
 Perpendicular correction:      1.0000E+00
Species  2, function  1, Maxwellian fit: 
 Initial fit parameter  1 =     1.4128E+04
 Initial fit parameter  2 =     1.8360E+03
 Initial fit parameter  3 =     0.0000E+00
 Initial fit parameter  4 =     0.0000E+00
 Initial fit parameter  5 =     0.0000E+00
 Perpendicular correction:      1.8360E+03
-=-=-=-=-=-=-=-=-
Scan 1: kpar from    1.000E-03 to    1.000E-02
-=-=-=-=-=-=-=-=-
Attempting to read f0 array from file: test_kpar_fast
-=-=-=-=-=-=-=-=-
Calculating df0/dpperp, df0/ppar if necessary...
 Derivatives calculated
-=-=-=-=
Species   1:
 Integration:                  9.9958E-01
 Charge density:               9.9958E-01
 Parallel current density:     5.8205E-18
-=-=-=-=
Species   2:
 Integration:                  9.9958E-01
 Charge density:              -9.9958E-01
 Parallel current density:     6.7357E-16
-=-=-=-=
 Total charge density:              -1.6875E-14
 Total parallel current density:     6.7939E-16
-=-=-=-=-=-=-=-=
-=-=-=-=-=-=-=-=-
Determine if relativistic calculation necessary
No relativistic calculation necessary.
-=-=-=-=-=-=-=-=-
Determine fit parameters for hybrid analytic continuation...
 Results of the fit for hybrid analytic continuation at iperp = 1:
  param_fit( 1, 1, 1, 1) =     1.7959E-01
  param_fit( 1, 1, 2, 1) =     1.0000E+00
  param_fit( 1, 1, 3, 1) =    -3.7522E-18
 
  param_fit( 2, 1, 1, 1) =     1.4128E+04
  param_fit( 2, 1, 2, 1) =     1.8360E+03
  param_fit( 2, 1, 3, 1) =     2.1967E-19
 
 Sum of all least-squares:     3.3601E-08
 Standard error of the estimate:     7.6378E-07
 Writing fit result to files
-=-=-=-=
Species   1:
 Integration of fit/analytical function:                  9.9958E-01
 Charge density of fit/analytical function:               9.9958E-01
 Parallel current density of fit/analytical function:    -2.6211E-17
-=-=-=-=
Species   2:
 Integration of fit/analytical function:                  9.9958E-01
 Charge density of fit/analytical function:              -9.9958E-01
 Parallel current density of fit/analytical function:    -6.3918E-16
-=-=-=-=
 Total charge density of fit/analytical function:               4.5757E-07
 Total parallel current density of fit/analytical function:    -6.6539E-16
-=-=-=-=-=-=-=-=
-=-=-=-=-=-=-=-=-
Broadcasting df0/dp...
 df0/dp received
Required nMax for species 1 : 21
Required nMax for species 2 : 13
Ideal Number of Processors: 36
-=-=-=-=-=-=-=-=-=-
Processor    0 of    4 ready. Species    0: n in [   0   0]
Processor    2 of    4 ready. Species    1: n in [  11  21]
Processor    3 of    4 ready. Species    2: n in [   0  13]
Processor    1 of    4 ready. Species    1: n in [   0  10]
Starting Secant Method
 Refining Roots:
Root 1
 Converged after iteration    3
 D(    9.9891E-04    4.3032E-09)=     8.8808E-16   -9.6238E-15
   1    9.9891E-04    4.3032E-09   -1.4015E+01    8.8808E-16   -9.6238E-15
 => solution/test_kpar_fast.scan_kpara_1.root_1
 => solution/test_kpar_fast.eigen_kpara_1.root_1
 => solution/test_kpar_fast.heat_kpara_1.root_1
kperp:     1.0000E-02 kpar:     1.0746E-03
 Converged after iteration    5
 D(    1.0734E-03   -6.4373E-09)=     7.8756E-15    2.1660E-15
kperp:     1.0000E-02 kpar:     1.1548E-03
 Converged after iteration    5
 D(    1.1535E-03    8.0137E-10)=    -1.9329E-15   -7.5235E-15
kperp:     1.0000E-02 kpar:     1.2409E-03
 Converged after iteration    5
 D(    1.2396E-03   -7.3642E-09)=     1.0496E-14    2.8373E-15
kperp:     1.0000E-02 kpar:     1.3335E-03
 Converged after iteration    5
 D(    1.3321E-03    9.0951E-10)=    -2.5908E-15   -9.9421E-15
kperp:     1.0000E-02 kpar:     1.4330E-03
 Converged after iteration    5
 D(    1.4314E-03   -8.3969E-09)=     1.3990E-14    3.6899E-15
kperp:     1.0000E-02 kpar:     1.5399E-03
 Converged after iteration    5
 D(    1.5382E-03    3.8511E-09)=    -5.8534E-15   -1.7981E-14
kperp:     1.0000E-02 kpar:     1.6548E-03
 Converged after iteration    5
 D(    1.6530E-03   -9.5349E-09)=     1.8642E-14    4.7535E-15
kperp:     1.0000E-02 kpar:     1.7783E-03
 Converged after iteration    5
 D(    1.7763E-03   -1.4789E-09)=    -2.2988E-15   -1.4044E-14
kperp:     1.0000E-02 kpar:     1.9110E-03
 Converged after iteration    5
 D(    1.9089E-03    1.1806E-09)=    -5.4254E-15   -1.9620E-14
kperp:     1.0000E-02 kpar:     2.0535E-03
 Converged after iteration    5
 D(    2.0513E-03   -1.1418E-08)=     2.8654E-14    6.7678E-15
kperp:     1.0000E-02 kpar:     2.2067E-03
 Converged after iteration    5
 D(    2.2043E-03    1.2713E-09)=    -7.3114E-15   -2.5524E-14
kperp:     1.0000E-02 kpar:     2.3714E-03
 Converged after iteration    5
 D(    2.3688E-03   -1.2758E-08)=     3.8128E-14    8.3334E-15
kperp:     1.0000E-02 kpar:     2.5483E-03
 Converged after iteration    5
 D(    2.5455E-03    5.6579E-09)=    -1.6439E-14   -4.5381E-14
kperp:     1.0000E-02 kpar:     2.7384E-03
 Converged after iteration    5
 D(    2.7354E-03   -1.2019E-08)=     4.2417E-14    2.6814E-15
kperp:     1.0000E-02 kpar:     2.9427E-03
 Converged after iteration    5
 D(    2.9395E-03    1.1599E-09)=    -1.3364E-14   -4.1819E-14
kperp:     1.0000E-02 kpar:     3.1623E-03
 Converged after iteration    5
 D(    3.1588E-03   -1.5471E-08)=     6.7193E-14    1.1056E-14
kperp:     1.0000E-02 kpar:     3.3982E-03
 Converged after iteration    5
 D(    3.3945E-03    6.0740E-09)=    -2.9768E-14   -7.2658E-14
kperp:     1.0000E-02 kpar:     3.6517E-03
 Converged after iteration    5
 D(    3.6477E-03   -1.6772E-08)=     8.8797E-14    1.0989E-14
kperp:     1.0000E-02 kpar:     3.9242E-03
 Converged after iteration    5
 D(    3.9199E-03   -3.0511E-10)=    -2.6319E-14   -7.6512E-14
kperp:     1.0000E-02 kpar:     4.2170E-03
 Converged after iteration    5
 D(    4.2123E-03   -6.0924E-09)=    -1.8120E-14   -6.0117E-14
kperp:     1.0000E-02 kpar:     4.5316E-03
 Converged after iteration    5
 D(    4.5266E-03   -7.5320E-09)=    -2.1793E-14   -6.5813E-14
kperp:     1.0000E-02 kpar:     4.8697E-03
 Converged after iteration    5
 D(    4.8643E-03   -9.5271E-09)=    -2.6171E-14   -7.1322E-14
kperp:     1.0000E-02 kpar:     5.2330E-03
 Converged after iteration    5
 D(    5.2272E-03   -2.1552E-08)=    -2.1363E-14   -2.0522E-14
kperp:     1.0000E-02 kpar:     5.6234E-03
 Converged after iteration    5
 D(    5.6172E-03   -2.4803E-08)=    -2.5528E-14   -2.4754E-14
kperp:     1.0000E-02 kpar:     6.0430E-03
 Converged after iteration    5
 D(    6.0362E-03   -2.9724E-08)=    -3.5507E-14   -2.8086E-14
kperp:     1.0000E-02 kpar:     6.4938E-03
 Converged after iteration    5
 D(    6.4865E-03   -3.5835E-08)=    -3.7356E-14   -3.7646E-14
kperp:     1.0000E-02 kpar:     6.9783E-03
 Converged after iteration    5
 D(    6.9704E-03   -4.5260E-08)=    -4.6645E-14   -4.7236E-14
kperp:     1.0000E-02 kpar:     7.4989E-03
 Converged after iteration    5
 D(    7.4904E-03   -5.9513E-08)=    -6.3129E-14   -5.6164E-14
kperp:     1.0000E-02 kpar:     8.0584E-03
 Converged after iteration    5
 D(    8.0492E-03   -8.6912E-08)=    -1.7326E-14   -1.8930E-15
kperp:     1.0000E-02 kpar:     8.6596E-03
 Converged after iteration    6
 D(    8.6496E-03   -1.1826E-07)=    -1.1804E-16    5.9983E-17
kperp:     1.0000E-02 kpar:     9.3057E-03
 Converged after iteration    6
 D(    9.2948E-03   -1.6390E-07)=    -6.4627E-17    4.3255E-17
kperp:     1.0000E-02 kpar:     1.0000E-02
 Converged after iteration    6
 D(    9.9881E-03   -2.3049E-07)=    -4.4331E-17    5.8770E-17
Finishing ALPS===================================
Time:
2023-07-20 -- 15:24:49
================================================
mahmoudibrahim@dhcp12 ALPS-master % 
Thanks!
Mahmoud

**Reply by danielver02** [reply]
This means the code is running well. I'm not sure why the test suite thinks otherwise. It's definitely running correctly here. Maybe it has to do with the shell behaviour when you execute the test script? Not sure. Anyways, the important thing is that the code runs as expected.

**Reply by MSAfify** [reply]
Great thanks, I wish you a very nice afternoon!
Mahmoud

**Comment by MSAfify** [comment]
Hi Prof Daniel,
I wonder if the plotting routines for ALPS results are available or not on Github.
Thanks!
Mahmoud

**Reply by danielver02** [reply]
Hi Mahmoud,
No, we do not include plotting routines for the ALPS results. ALPS output is done in simple ASCII files, and we recommend using your preferred visualisation software (Python, Matlab, gnuplot, etc.) for the plotting.
Best wishes,
-Daniel

**Comment by MSAfify** [comment]
Hi,
Great thanks!
Mahmoud

**Comment by MSAfify** [comment]
Hi Prof. Daniel,
Please let me know what the difference is between the following two distribution files: test_map_dist.in and test_kperp_dist.in. It seems the files have similar contents. Such distribution files are bi-Kappa, right?
Another inquiry, please for each type of the distribution files, it is possible to change the number of species, and other parameters could be changed via the associated input file for each type of the distribution files, so how can I consider different types of distributions for the same dispersion relation?
Thanks!
Mahmoud

**Reply by danielver02** [reply]
Hi Mahmoud,

Both distribution files are basically identical. They are simply used by two different ALPS input routines in the test directory, and for better clarity, we have given each of them their own distribution input files. The distributions in both cases are simple, isotropic Maxwellian distributions.

You can create distribution files for an arbitrary number of species. They follow the same naming convention (.1.array, .2.array, .3.array, etc.). In the ALPS run file, you can then choose the number of species (through the nspec) parameter. The code will then automatically look for the corresponding .array files in the ./distribution folder. Remember though to add namelist blocks for each species in the ALPS run files, following the format of the standard files. In those blocks, you would define the charge, mass, density, etc. of each species.

Best wishes,
-Daniel

P.S. - I would recommend opening new discussion topics when you have new questions. This makes it a bit easier for us to follow your ideas and to address them. At the moment, they're all in one thread following the general welcome message. Thank you!

**Reply by MSAfify** [reply]
Hi,
Ok, thanks so much!
Mahmoud
